sklearn/metrics/tests/test_classification.py

```diff
@@ -2905,3 +2905,216 @@ def test_brier_score_loss_deprecation_warning():
             y_prob=y_pred,
             y_proba=y_pred,
         )
+
+
+def test_d2_log_loss_score():
+    y_true = [0, 0, 0, 1, 1, 1]
+    y_pred = np.array(
+        [
+            [0.5, 0.5],
+            [0.9, 0.1],
+            [0.4, 0.6],
+            [0.6, 0.4],
+            [0.35, 0.65],
+            [0.01, 0.99],
+        ]
+    )
+    y_pred_null = np.array(
+        [
+            [0.5, 0.5],
+            [0.5, 0.5],
+            [0.5, 0.5],
+            [0.5, 0.5],
+            [0.5, 0.5],
+            [0.5, 0.5],
+        ]
+    )
+    d2_score = d2_log_loss_score(y_true, y_pred)
+    log_likelihood = log_loss(y_true=y_true, y_pred=y_pred)
+    log_likelihood_null = log_loss(y_true=y_true, y_pred=y_pred_null)
+    d2_score_true = 1 - log_likelihood / log_likelihood_null
+    assert_almost_equal(d2_score, d2_score_true)
+
+    # check if good predictions give a relatively higher value for the d2 score
+    y_true = [0, 0, 0, 1, 1, 1]
+    y_pred = np.array(
+        [
+            [0.9, 0.1],
+            [0.8, 0.2],
+            [0.9, 0.1],
+            [0.1, 0.9],
+            [0.2, 0.8],
+            [0.1, 0.9],
+        ]
+    )
+    d2_score = d2_log_loss_score(y_true, y_pred)
+    assert 0.5 < d2_score < 1.0
+
+    # check if poor predictions gives a relatively low value for the d2 score.
+    y_true = [0, 0, 0, 1, 1, 1]
+    y_pred = np.array(
+        [
+            [0.5, 0.5],
+            [0.1, 0.9],
+            [0.1, 0.9],
+            [0.9, 0.1],
+            [0.75, 0.25],
+            [0.1, 0.9],
+        ]
+    )
+    d2_score = d2_log_loss_score(y_true, y_pred)
+    assert d2_score < 0
+
+    # check if simply using the average of the classes as the predictions
+    # gives a d2 score of 0
+    y_true = [0, 0, 0, 1, 1, 1]
+    y_pred = np.array(
+        [
+            [0.5, 0.5],
+            [0.5, 0.5],
+            [0.5, 0.5],
+            [0.5, 0.5],
+            [0.5, 0.5],
+            [0.5, 0.5],
+        ]
+    )
+    d2_score = d2_log_loss_score(y_true, y_pred)
+    assert d2_score == 0
+
+    # check if simply using the average of the classes as the predictions
+    # gives a d2 score of 0 when the positive class has a higher proportion.
+    y_true = [0, 1, 1, 1]
+    y_pred = np.array([[0.25, 0.75], [0.25, 0.75], [0.25, 0.75], [0.25, 0.75]])
+    d2_score = d2_log_loss_score(y_true, y_pred)
+    assert d2_score == 0
+
+
+def test_d2_log_loss_score_with_y_true_as_strings():
```

Comment:
I think we can remove this test and just add string labelled versions to  test_d2_log_loss_score, e.g.
```python
    # check if good predictions give a relatively higher value for the d2 score
    y_true = [0, 0, 0, 1, 1, 1]
    y_pred = np.array(
        [
            [0.9, 0.1],
            [0.8, 0.2],
            [0.9, 0.1],
            [0.1, 0.9],
            [0.2, 0.8],
            [0.1, 0.9],
        ]
    )
    d2_score = d2_log_loss_score(y_true, y_pred)
    assert 0.5 < d2_score < 1.0
    # same for string labels
    y_true = ["black", "black", "black", "white", "white", "white"]
    d2_score_2 = d2_log_loss_score(y_true, y_pred)
    assert d2_score_2 == pytest.approx(d2_score)
```

Reply:
Okay I got it now. Thanks.